### PR TITLE
Script Operator Enhancements RFC

### PR DESCRIPTION
**Pull request by Licenser**
[Rendered](https://github.com/tremor-rs/tremor-rfcs/blob/script-operator-enhancements/text/0000-script-operator-enhacements.md)

# Checklist

* [x] Add a link to rendered text (see above)
* [x] Assign core group reviewer (manual: assign review)
* [x] Mark as ready for review / non-draft when ready for open review (manual: move to open for review)
* [ ] Core reviewer accepts the once the open review period is over, and there are no more outstanding issues (automation: accepted)
* [ ] Assign RFC number by the end of the review period

### CODE REVIEW

File: text/0000-script-operator-enhacements.md [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+
+# Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+This introduces two new parts to the to the seelct statement.
+
+## `state`
+
+The `state` section is introduced as an optional section to provide an initial stat. One alternative would be calling it `init` however using that would introduce a new keyword and by that break backwards compatibility and reduce the number of possible idents.
```
Comment (darach) [inline text/0000-script-operator-enhacements.md:223]: `stat` -> `state`

File: text/0000-script-operator-enhacements.md [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+
+[summary]: #summary
+
+This RFC aims to add quality of life improvements to the script operator. As it exists today, the operator works well for simple use-cases but-in light of the growing number of complex pipelines and the use of patterns like the configurator pattern it is cumbersome. When first implemented, `state` didn't exist, and ports were not used. Those are the areas of improvement this RFC tackles.
+
+# Motivation
+
+[motivation]: #motivation
+
+To demonstrate I will give a motivating example if the reduction of complexity for a simplified algorithm for distributed loadbalancing:
```
Comment (darach) [inline text/0000-script-operator-enhacements.md:16]: I -> we

File: text/0000-script-operator-enhacements.md [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+
+The original code has to fight with not knowing what ports data comes in and having to verify the initalization of state on every event.
```
Comment (darach) [inline text/0000-script-operator-enhacements.md:18]: fight -> contend

**Comment by Licenser** [comment]
adding a quick note from Jonathan that we using both `args` and `with` for script args

**Comment by Licenser** [comment]
migrated to https://github.com/tremor-rs/tremor-www/pull/120

**Pull request closed without merging**